### PR TITLE
Invalid WKT format query fail as INVALID_FUNCTION_ARGUMENT

### PR DESCRIPTION
**Pull request by Lewuathe**
Some Geospatial functions with invalid WKT input fails dues to `INTERNAL_ERROR`.

```
java.lang.IllegalArgumentException: undefined
	at com.esri.core.geometry.WktParser.lineStringStart_(WktParser.java:479)
	at com.esri.core.geometry.WktParser.nextToken(WktParser.java:87)
	at com.esri.core.geometry.OperatorImportFromWktLocal.polygonText(OperatorImportFromWktLocal.java:500)
	at com.esri.core.geometry.OperatorImportFromWktLocal.polygonTaggedText(OperatorImportFromWktLocal.java:179)
	at com.esri.core.geometry.OperatorImportFromWktLocal.importFromWkt(OperatorImportFromWktLocal.java:112)
	at com.esri.core.geometry.OperatorImportFromWktLocal.executeOGC(OperatorImportFromWktLocal.java:76)
	at com.esri.core.geometry.ogc.OGCGeometry.fromText(OGCGeometry.java:470)
```

This type of error is a kind of syntax error and also we can improve the error message by catching exceptions.

- [x] : Change error code from `INTERNAL_ERROR` to `INVALID_FUNCTION_ARGUMENT`
- [x] : Catch exception not to fail `IllegalArgumentException` with `undefined` message

see: #9511

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
@Lewuathe there are conflicts, can you please rebase?

File: presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java [hunk 1]
```diff
@@ -561,9 +561,14 @@ public static Boolean stWithin(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType
 
     private static OGCGeometry geometryFromText(Slice input)
     {
-        OGCGeometry geometry = OGCGeometry.fromText(input.toStringUtf8());
-        geometry.setSpatialReference(null);
-        return geometry;
+        try {
```
Comment (nezihyigitbasi) [inline presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java]: we can only wrap the code that can throw the `IllegalArgumentException`:

```java
OGCGeometry geometry;
try {
    geometry = OGCGeometry.fromText(input.toStringUtf8());
}
catch (IllegalArgumentException e) {
    throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid WKT: " + input.toStringUtf8(), e);
}
geometry.setSpatialReference(null);
return geometry;
```

(Note that this also changed the `throw` statement).

File: presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java [hunk 2]
```diff
@@ -475,4 +476,13 @@ public void testSTWithin()
         assertFunction("ST_Within(ST_GeometryFromText('LINESTRING (1 1, 3 3)'), ST_GeometryFromText('POLYGON ((0 0, 0 4, 4 4, 4 0))'))", BOOLEAN, true);
         assertFunction("ST_Touches(ST_GeometryFromText('MULTIPOLYGON (((1 1, 1 3, 3 3, 3 1)), ((0 0, 0 2, 2 2, 2 0)))'), ST_GeometryFromText('POLYGON ((0 1, 3 1, 3 3, 0 3))'))", BOOLEAN, false);
     }
+
+    @Test
+    public void testInvalidWKT()
+            throws Exception
+    {
+        assertInvalidFunction("ST_LineFromText('LINESTRING (0 0, 1)')", StandardErrorCode.INVALID_FUNCTION_ARGUMENT, "invalid WKT form: LINESTRING (0 0, 1)");
```
Comment (nezihyigitbasi) [inline presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java]: With the change above the test should be updated too. Please also static import `INVALID_FUNCTION_ARGUMENT`.

**Comment by nezihyigitbasi** [comment]
@Lewuathe I made these changes and merged this PR, thanks for your contribution!

**Comment by Lewuathe** [comment]
@nezihyigitbasi Thank you for taking care!

**Pull request closed without merging**